lib/sql-string.js

```diff
@@ -193,14 +193,16 @@ SqlString.bufferToString = function(buffer, dialect) {
 };
 
 SqlString.objectToValues = function(object, timeZone) {
-  var values = [];
-  for (var key in object) {
```

Comment:
Well most important about performance is to prevent deoptimization that might occur with for in loops. I'll have a look at the code soon with http://mrale.ph/irhydra/2/ to check what's causing bad deopts in the code. If you want to compare loops have a look at: http://jsperf.com/fastest-array-loops-in-javascript/371
Using a for in loop you'll have to check if the element is a own property and this destroys all readability opposed to the other solutions. Comparing it with the while pop loop it's even less readable in my opinion, since you have less indentation in the while loop.